src/Data/Module.purs

```diff
@@ -0,0 +1,50 @@
+module Data.Module where
+
+import Data.Ring (class Ring, zero, add, sub, mul)
+import Data.Unit (Unit, unit)
+
+-- | The `Module` class is for types that support addition as well as
+-- | multiplication by a scalar on the left.
+-- |
+-- | If you want a vector space, simply specify an additional `Field s`
+-- | constraint.
+-- |
+-- | Instances must satisfy the following laws in addition to the `Ring` laws:
+-- |
+-- | - Commutative monoid under addition:
+-- |   - Associativity: `(a $+ b) $+ c = a $+ (b $+ c)`
+-- |   - Identity: `mlzero $+ a = a $+ mlzero = a`
+-- |   - Commutative: `a $+ b = b $+ a`
+-- | - Multiplicative identity: `one $* a = a`
+-- | - Multiplication distributes over addition:
+-- |   -  distributivity: `a $* (b $+ c) = (a $* b) $+ (a $* c)`
+-- |   - Right distributivity: `(a + b) $* c = (a $* c) $+ (b $* c)`
+-- | - Annihilation: `zero $* a = mlzero`
+-- | - Compatibility: `(a * b) $* c = a $* (b $* c)`
+class (Ring s) <= Module m s | m -> s where
```

Comment:
`CommutativeRing`?

Reply:
Yes, good catch.

Reply:
I'd prefer `Module s m` so that `Module s` could be used as a constraint kind.